synapse/lib/layer.py

```diff
@@ -3611,6 +3617,8 @@ def _editPropDel(self, buid, form, edit, sode, meta):
                     if realtype == STOR_TYPE_NDEF:
                         self.layrslab.delete(indx, buid + abrv, db=self.byndef)
 
+                await asyncio.sleep(0)
```

Comment:
Should we wait to pop the prop and set the sode to dirty until after we've removed these index values to ensure we don't have index values pointing to incorrect nodes?